### ISSUE TITLE
warning screen should always be enabled or at the very minimum togglable

### ISSUE THREAD

**Issue by barbudreadmon** [libretro/mame#314]
Warning screen is an important feature of MAME, among other things it allows the user to know if a game is **playable**, which is primordial information if you don't want to waste your time desperately trying to launch a game that simply doesn't work.

For some weird reason, this important feature is disabled in the libretro port, and it doesn't seem there is any way to enable it. I think it has to do with the following code : 
https://github.com/libretro/mame/blob/fdb7edfb0d142ddb93aa558daa6a771a40fe1c4e/src/frontend/mame/ui/ui.cpp#L418-L421

**Comment by riggles1** [comment]
Wouldn't be bad, sure the info is in the mame menu as well, but it helps newcomers from asking unnecessary things.

As long as we can set:
skip_gameinfo             1
skip_warnings              1

in the RA root folder/mame.ini file for us who set the core to read additional settings from the .ini.

**Comment by Tasosgemah** [comment]
Optional, sure. "Always" nope.

**Comment by ghost** [comment]
Just for the record the code has not been disabled anywhere. It not settings based why its not happening this one does need looked into.

**Comment by ghost** [comment]
There is nothing that can be done without hacking code up and is desirable to stay upstream. If they fix the EMSCRIPTEN port we can implement them fixes here.

**Comment by barbudreadmon** [comment]
>There is nothing that can be done without hacking code up and is desirable to stay upstream

The code is already hacked... It is because it is hacked that this message isn't shown anymore...

>If they fix the EMSCRIPTEN port

Seriously ? Are you trolling intentionally ?

**Comment by ghost** [comment]
sorry but its the same upstream for the same reason look here. 

https://github.com/mamedev/mame/blob/aefb991c77462a867f6cd3d16fe912374cc8772e/src/frontend/mame/ui/ui.cpp#L422-L425

This is ra limitations the way its run model works sorry if you dont like the really of the matter. The main point to take from this is no one has intentionally disabled them which was unclear at the start of the topic.

**Comment by barbudreadmon** [comment]
>sorry but its the same upstream for the same reason look here.

This is not emscripten here, this is the libretro project.
This message existed in earlier versions of this core. 
This message still exists in some vintaged mame cores, and can be toggled through core options.
It was intentionally disabled by someone, this decision was even criticized by the mame team for legit reasons.

Saying it will be restored when emscripten will be fixed (meaning never), while the 2 things are totally unrelated, is yet another non-sense.

**Comment by ghost** [comment]
Well it is open source if you  or anyone else want to put a pull request in feel free or anyone else can. Im giving you a legitimate why they arent there. If you feel strongly contract the mame team about the emscripten port not having the info screens. I dont know what to say to you beyond that.  I wont get drawn into your baseless accusations, if you disable that define mame simply wont start.

**Comment by sonninnos** [comment]
Yep, had a look at this, and it isn't that simple to allow those button waiting startup screens since they start waiting before the retro loop starts running, so it needs much heavier modifications with questionable payoffs. Maybe the worst "this game won't work" warnings coud at least use the non-blocking libretro message system.
